### PR TITLE
doc: add style guide

### PR DESCRIPTION
**Pull request by Wodann**
Proposal for adding guidelines around code style

### CODE REVIEW

File: book/src/02_development/06_style_guide.md [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+#[derive(PartialEq, Eq, PartialOrd, Ord)]
+```
+
+## Member ordering
+
+When adding a variant to an `enum` or a field to a `struct` or enum variant, by default follow alphabetical order. If it makes more sense to follow custom ordering, feel free to do so.
+
+### Member functions
+
+For member functions, use the following default rules to order them:
```
Comment (agostbiro) [inline book/src/02_development/06_style_guide.md:31]: "default rules" -- I'm assuming this means it's ok to deviate if it makes sense? Would be good to make it more explicit if possible

**Review by agostbiro — APPROVED** [review]
Thanks for this! I left one comment, but approving to unblock merging

File: book/src/02_development/06_style_guide.md [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+This is a style guide for the EDR project.
+
+## Procedural derive macros
+
+When deriving multiple traits, use the following rules to order them:
+
+1. Standard library traits before external crates
+2. Sub-traits before super-traits
+3. Alpabetical order
```
Comment (Xanewok) [inline book/src/02_development/06_style_guide.md:11]:
```suggestion
3. Alphabetical order
```

**Review by Xanewok — APPROVED** [review]

File: book/src/02_development/06_style_guide.md (hunk 1 above)
Comment (Wodann) [reply book/src/02_development/06_style_guide.md:31]: Addressed in https://github.com/NomicFoundation/edr/pull/660/commits/d1a250f6ebd06ebdf3d9ebeaf7c172e057135d54

**Review by fvictorio — APPROVED** [review]

**Comment by CindaiMKR** [comment]
$ npx hardhat node
Started HTTP and WebSocket JSON-RPC server at http://127.0.0.1:8545/

Accounts
========
Account #0: 0xf39fd6e51aad88f6f4ce6ab8827279cfffb92266 (10000 ETH)
Private Key: 0xac0974bec39a17e36ba4a6b4d238ff944bacb478cbed5efcae784d7bf4f2ff80

Account #1: 0x70997970c51812dc3a010c7d01b50e0d17dc79c8 (10000 ETH)
Private Key: 0x59c6995e998f97a5a0044966f0945389dc9e86dae88c7a8412f4603b6b78690d
...

**Pull request merged**